### PR TITLE
[Bootstrap] Refactoring to remove autoloading dynamic dataobject classes

### LINKED ISSUE #14485: Refactor Bootstrap: remove autoloading for dynamic dataobject classes
### Task description
Related to https://github.com/pimcore/pimcore/issues/10344

- [x] Remove including autoload files in `bootstrap()` after https://github.com/pimcore/skeleton/pull/128/files is merged.
- [x]  Remove `setAutoloader`, `getAutoloader` & `autoload` methods from Bootstrap.php as these methods are used to dynamically autoload dataobject classes from var directory(or custom var directory) which should be handled in project `composer.json` (add docs [here](https://github.com/pimcore/pimcore/blob/543457a81d2171174f277677ea7e20e25b7ff6c3/doc/Development_Documentation/01_Getting_Started/03_Configuration.md#L63)).
- [x] Remove unused constant `PIMCORE_APP_BUNDLE_CLASS_FILE`

### PR DESCRIPTION
**Pull request by dvesh3**
## Changes in this pull request  
Resolves #14485

### CODE REVIEW

File: doc/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md [hunk 1]
```diff
@@ -237,6 +237,9 @@ pimcore:
             - en
             - de
 ```
+- [Bootstrap]
+  - Relying on `Pimcore\Bootstrap::bootstrap()` for autoloading classes will not work anymore. 
+  - Removed unused constant `PIMCORE_APP_BUNDLE_CLASS_FILE`
```
Comment (kingjia90) [inline doc/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md:249]: just noticed this leftover https://github.com/pimcore/pimcore/blob/58cd6f66bd7483d429a80f4c4d5739ea537585b0/stubs/dynamic-constants.php#L22

File: lib/Bootstrap.php [hunk 2]
```diff
@@ -198,19 +183,6 @@ public static function defineConstants(): void
         $resolveConstant('PIMCORE_KERNEL_CLASS', '\App\Kernel');
     }
 
-    private static function autoload(): void
-    {
-        $loader = \Pimcore::getAutoloader();
-
-        // tell the autoloader where to find Pimcore's generated class stubs
-        // this is primarily necessary for tests and custom class directories, which are not covered in composer.json
```
Comment (kingjia90) [inline lib/Bootstrap.php:206]: by reading this comment. would this PR affect tests of some bundle that generate class stubs or so?
https://github.com/pimcore/pimcore/blob/58cd6f66bd7483d429a80f4c4d5739ea537585b0/tests/_bootstrap.php#L45-L47

https://github.com/pimcore/pimcore/blob/58cd6f66bd7483d429a80f4c4d5739ea537585b0/phpstan-bootstrap.php#L19-L21

**Review by kingjia90 — COMMENTED** [review]
LGTM but i would like to have a second look by someone more into the topic 🤔 not sure if i am missing something

File: doc/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md (hunk 1 above)
Comment (dvesh3) [reply doc/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md:249]: Done

File: lib/Bootstrap.php (hunk 2 above)
Comment (dvesh3) [reply lib/Bootstrap.php:206]: Not really, as we autoload them in tests bootstrap with `Autoloader::addNamespace('Pimcore\Model\DataObject', PIMCORE_CLASS_DIRECTORY . '/DataObject');`

**Pull request merged**